File: solutions/java/resistor-color/1/src/main/java/ResistorColor.java
import java.util.List;
import java.util.Arrays;

class ResistorColor {
    private final String[] COLORS_ARRAY = {"black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"};
    private final List<String> COLORS_LIST = Arrays.asList(COLORS_ARRAY);
    
    int colorCode(String color) {
        return COLORS_LIST.indexOf(color);
    }

    String[] colors() {
        return this.COLORS_ARRAY;
    }
}

File: solutions/python/resistor-color/1/resistor_color.py
RESISTOR_COLOR_DICT = {
    "black": 0,
    "brown": 1,
    "red": 2,
    "orange": 3,
    "yellow": 4,
    "green": 5,
    "blue": 6,
    "violet": 7,
    "grey": 8,
    "white": 9,
}


def color_code(color):
    return RESISTOR_COLOR_DICT.get(color)


def colors():
    return list(RESISTOR_COLOR_DICT.keys())
